catalog/transformer/catalog-transformer-xml/src/main/java/ddf/catalog/transformer/xml/adapter/AdaptedMetacard.java

```diff
@@ -188,11 +189,11 @@ public final Attribute getAttribute(String name) {
         if (Metacard.ID.equals(name)) {
             return this.id;
         }
-        for (Attribute attribute : attributes) {
-            if (attribute == null || StringUtils.isEmpty(attribute.getName())) {
-                continue;
-            } else if (name.equals(attribute.getName())) {
-                return attribute;
+        for (Attribute attribute : Arrays.asList((attributes.toArray(new Attribute[attributes.size()])))) {
```

Comment:
Also the `asList()` isn't necessary since you can just iterate over the array directly.